ui/src/tool-bar/tool-bar.tsx

```diff
@@ -622,12 +623,16 @@ export const ToolBar: React.FC<ToolbarParams> = ({
                     onCancel={() => setShowEditWorkspaceDialog(false)}
                     onSave={(_, request) => {
                         if (currentDesign) {
-                            updateLayoutDesign(currentDesign.id, request).finally(() => {
-                                updateLayoutDesignChangeTime();
-                                setShowEditWorkspaceDialog(false);
-                            });
+                            setSavingWorkspace(true);
+                            updateLayoutDesign(currentDesign.id, request)
+                                .finally(() => {
+                                    updateLayoutDesignChangeTime();
+                                    setShowEditWorkspaceDialog(false);
+                                })
+                                .finally(() => setSavingWorkspace(false));
```

Comment:
Kaksi .finally-kutsua ketjussa näyttää omaan silmään hieman hämärältä 🤔

Reply:
Ja mitäs muuten tässä nyt käykään dialogille jos bäkkäriltä lentää virhe 🤔

Reply:
Tuo kaksi `.finally()`-kutsua putkeen on fiba, mutta sen ei pitäisi vaikuttaa toimintaan. Korjailen tuon toisen tiketin yhteydessä.

Bäkkärivirheen sattuessa napit enabloituvat uudestaan. Tämä on meillä aika yleinen patterni, sillä kyse voi olla esim. verkkoräpsystä, jolloin uusi yritys saattaa korjata asian.

Reply:
Mjees, lähinnä mietin ettei kai se dialogi nyt katoaisi jos bäkkäri palauttaa virhettä -- toiminnee siis ideaalisesti

Reply:
Itse asiassa nyt kun sanoit niin kyllähän se katoaa 🤔 Nuo aiemmat ehdot pitäisi todennäköisesti tehdä `.then()`:in sisällä, ei `.finally()`:ssä. Korjaanpas vielä sen